### PR TITLE
Added a few missing unit tests

### PR DESCRIPTION
**Pull request by ernstsson**

### CODE REVIEW

File: agera/src/test/java/com/google/android/agera/ObservablesTest.java [hunk 1]
```diff
@@ -325,6 +325,32 @@ public void shouldUpdatePerMillisecondObservable() {
...
+  public void shouldUpdateCompositeOfPerMillisecondObservable() {
+    final long expectedDelayedTime = scheduler.getCurrentTime() + FILTER_TIME;
+    updatable.addToObservable(compositeObservable(perMillisecondObservable(
+        FILTER_TIME, updateDispatcher), updateDispatcher()));
+
+    updateDispatcher.update();
+    idleMainLooper(FILTER_TIME);
+
+    assertThat(updatable, wasUpdated());
+    assertThat(scheduler.getCurrentTime(), greaterThanOrEqualTo(expectedDelayedTime));
```
Comment (smile616) [inline agera/src/test/java/com/google/android/agera/ObservablesTest.java]: it check that idleMainLooper(FILTER_TIME); works, doesn't it? why do you need it here?

File: agera/src/test/java/com/google/android/agera/ObservablesTest.java [hunk 2]
```diff
@@ -325,6 +325,32 @@ public void shouldUpdatePerMillisecondObservable() {
   }
 
   @Test
+  public void shouldUpdateCompositeOfPerMillisecondObservable() {
+    final long expectedDelayedTime = scheduler.getCurrentTime() + FILTER_TIME;
+    updatable.addToObservable(compositeObservable(perMillisecondObservable(
+        FILTER_TIME, updateDispatcher), updateDispatcher()));
+
+    updateDispatcher.update();
```
Comment (smile616) [inline agera/src/test/java/com/google/android/agera/ObservablesTest.java]: shouldn't it be something like:
`updateDispatcher.update();
idleMainLooper(FILTER_TIME / 2);
updateDispatcher.update();
updateDispatcher.update();
idleMainLooper(FILTER_TIME / 2);
assertThat(updatable, wasUpdatedOnce());
`

File: agera/src/test/java/com/google/android/agera/ObservablesTest.java [hunk 3]
```diff
@@ -325,6 +325,32 @@ public void shouldUpdatePerMillisecondObservable() {
...
+  public void shouldUpdateCompositeOfSinglePerMillisecondObservable() {
+    final long expectedDelayedTime = scheduler.getCurrentTime() + FILTER_TIME;
+    updatable.addToObservable(compositeObservable(perMillisecondObservable(
+        FILTER_TIME, updateDispatcher)));
+
+    updateDispatcher.update();
+    idleMainLooper(FILTER_TIME);
+
+    assertThat(updatable, wasUpdated());
+    assertThat(scheduler.getCurrentTime(), greaterThanOrEqualTo(expectedDelayedTime));
```
Comment (smile616) [inline agera/src/test/java/com/google/android/agera/ObservablesTest.java]: ditto

**Comment by ernstsson** [comment]
The intention was to make sure that a different composition of `perMillisecondObservable` still worked, meaning the construction is different but the assertion is the same as `shouldUpdatePerMillisecondObservable`.
As you pointed out, turns out the assertion in `shouldUpdatePerMillisecondObservable` is not correct. Looking into it.

**Pull request merged**